src/openms/source/FORMAT/FASTAFile.cpp

```diff
@@ -46,172 +46,192 @@
 
 namespace OpenMS
 {
-  using namespace std;
-  typedef seqan::RecordReader<std::fstream, seqan::SinglePass<> > FASTARecordReader;
-
-  FASTAFile::FASTAFile()
-  : reader_(std::nullptr_t()), // point to nothing
-    entries_read_(0)
-  {
-  }
-  
-  FASTAFile::~FASTAFile()
-  {
-    // infile_ and outfile_ will close automatically when going out of scope. No need to do it explicitly here.
-  }
-
-  void FASTAFile::readStart(const String& filename)
-  {
-    if (!File::exists(filename))
+    using namespace std;
+
+    FASTAFile::FASTAFile()
+            : entries_read_(0)
     {
-      throw Exception::FileNotFound(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
     }
 
-    if (!File::readable(filename))
+    FASTAFile::~FASTAFile()
     {
-      throw Exception::FileNotReadable(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+        // infile_ and outfile_ will close automatically when going out of scope. No need to do it explicitly here.
     }
 
-    if (infile_.is_open()) infile_.close(); // precaution
-
-    infile_.open(filename.c_str(), std::ios::binary | std::ios::in);
-
-    // Skip the header of PEFF files (http://www.psidev.info/peff)
-    std::string line;
-    std::streampos firstline = 0;
-    while (TextFile::getLine(infile_, line))
+    bool FASTAFile::readSequence(std::string & seq)
     {
-      if (!line.empty() && line[0] != '#')
-      {
-        break;
-      }
-      firstline = infile_.tellg();
-    }
-    infile_.seekg(firstline);
-  
-    // automatically deletes old handles
-    reader_ = std::unique_ptr<void, std::function<void(void*) > >(new FASTARecordReader(infile_),
-      [](void* ptr)
-      { // lambda with custom cast
-      delete static_cast<FASTARecordReader*>(ptr);
-      });
-
-    entries_read_ = 0;
-  }
-
-  bool FASTAFile::readNext(FASTAEntry& protein)
-  {
-    if (seqan::atEnd(*static_cast<FASTARecordReader*>(reader_.get())))
-    { 
-      // do NOT close(), since we still might want to seek to certain positions
-      return false;
+       std::string line;
+        while(TextFile::getLine(infile_, line) && line[0] != '>')
+        {
+            seq+=line;
+        }
+        nextID_=line;//because while loop stops AFTER reading first line of next protein
```

Comment:
the approach taken here will not work with external tools, which assume that `setPosition` and `position` can be used to jump to a position in the file and upon calling `readNext()`, the next sequence is obtained and the file position is now pointing to the next entry.
However, the approach taken here implies that after `readNext()`, the file position already points to the start of the sequence (not the start of the `>...` line).